src/gui/editors/PianoRoll.cpp

```diff
@@ -2873,214 +2878,257 @@ void PianoRoll::paintEvent(QPaintEvent * pe )
+		// if resizing the note edit area
+		if (m_action != ActionResizeNoteEditArea && partialKeyVisible != 0)
 		{
-			// draw a small one while checking if it is pressed or not
-			if( hasValidPattern() && m_pattern->instrumentTrack()->pianoModel()->isKeyPressed( key ) )
-			{
-				p.drawPixmap(PIANO_X, y - m_whiteKeySmallHeight, WHITE_KEY_WIDTH, m_whiteKeySmallHeight,
-							*s_whiteKeySmallPressedPm);
-			}
-			else
-			{
-				p.drawPixmap(PIANO_X, y - m_whiteKeySmallHeight, WHITE_KEY_WIDTH, m_whiteKeySmallHeight,
-							*s_whiteKeySmallPm);
-			}
-			// update y-pos
-			y -= m_whiteKeySmallHeight;
+			// adding height always means we don't have to add keys
+			m_notesEditHeight += partialKeyVisible;
+		}
+		updatePositionLineHeight();
+		int x, q = quantization(), tick;
 
+		// draw vertical quantization lines
+		// If we're over 100% zoom, we allow all quantization level grids
+		if (m_zoomingModel.value() <= 3)
+		{
+			// we're under 100% zoom
+			// allow quantization grid up to 1/24 for triplets
+			if (q % 3 != 0 && q < 8) { q = 8; }
+			// allow quantization grid up to 1/32 for normal notes
+			else if (q < 6) { q = 6; }
+		}
+		auto xCoordOfTick = [=](int tick) {
+			return WHITE_KEY_WIDTH + (
+				(tick - m_currentPosition) * m_ppb / MidiTime::ticksPerBar()
+			);
+		};
+		p.setPen(lineColor());
+		for (tick = m_currentPosition - m_currentPosition % q,
+			x = xCoordOfTick(tick);
+			x <= width();
+			tick += q, x = xCoordOfTick(tick))
+		{
+			p.drawLine(x, keyAreaTop(), x, noteEditBottom());
 		}
-		else if( prKeyOrder[key % KeysPerOctave] == PR_WHITE_KEY_BIG )
+
+		// draw horizontal grid lines and piano notes
+		p.setClipRect(0, keyAreaTop(), width(), keyAreaBottom() - keyAreaTop());
+		int key_line_y = keyAreaTop();
+		int grid_line_y = keyAreaTop() + m_keyLineHeight - 1;
+		static const int yTextCorrections[KeysPerOctave] = {
+		/*   C      D      E   F      G      A      B   */
+		    -4, 0, -8, 0, -6, -4, 0, -8, 0, -8, 0, -6
+		};
+		// lambda function for returning height of key
+		auto keyHeight = [&](
+			const int key
+		) -> int
 		{
-			// draw a big one while checking if it is pressed or not
-			if( hasValidPattern() && m_pattern->instrumentTrack()->pianoModel()->isKeyPressed( key ) )
+			switch (prKeyOrder[key % KeysPerOctave])
 			{
-				p.drawPixmap(PIANO_X, y - m_whiteKeyBigHeight, WHITE_KEY_WIDTH, m_whiteKeyBigHeight,
-							*s_whiteKeyBigPressedPm);
+			case PR_WHITE_KEY_SMALL:
+				return m_whiteKeySmallHeight;
+			case PR_WHITE_KEY_BIG:
+				return m_whiteKeyBigHeight;
+			case PR_BLACK_KEY:
+				return m_blackKeyHeight;
 			}
-			else
+			return 0; // should never happen
+		};
+		// lambda function to draw a key + horizontal grid line
+		auto drawKey = [&](
+			const int key,
+			const int yt)
+		{
+			const int keyCode = key % KeysPerOctave;
+			const bool pressed = m_pattern->instrumentTrack()->pianoModel()->isKeyPressed(key);
+			const int yb = yt + keyHeight(key);
+			int kw, kh;
```

Comment:
The compiler warns they may be used uninitialized, while it's not. Those warnings will make CI builds to fail.

Reply:
Will be fixed in next commit.